apps/sms/js/activity_handler.js

```diff
@@ -125,21 +125,26 @@ var ActivityHandler = {
       threadId: null
     };
 
-    // try to get a thread from number
-    // if no thread, promise is rejected and we try to find a contact
-    return MessageManager.findThreadFromNumber(viewInfo.number).then(
-      function onResolve(threadId) {
-        viewInfo.threadId = threadId;
-      },
-      function onReject() {
-        return ActivityHandler._findContactByNumber(viewInfo.number)
-          .then( (contact) => viewInfo.contact = contact);
-      }
-    )
-    // case no contact and no thread id: gobble the error
-    .catch(() => {})
-    // finally call toView whatever contact and threadId we have.
-    .then( () => this.toView(viewInfo) );
+    var threadPromise;
+    if (viewInfo.number) {
+      // try to get a thread from number
+      // if no thread, promise is rejected and we try to find a contact
+      threadPromise = MessageManager.findThreadFromNumber(viewInfo.number).then(
+        function onResolve(threadId) {
+          viewInfo.threadId = threadId;
+        },
+        function onReject() {
+          return ActivityHandler._findContactByNumber(viewInfo.number)
+            .then((contact) => viewInfo.contact = contact);
+        }
+      )
+      // case no contact and no thread id: gobble the error
+      .catch(() => {});
+    }
+
+    return (threadPromise || Promise.resolve()).then(
+      () => this.toView(viewInfo, !!viewInfo.threadId /* focusComposer */)
```

Comment:
Yeah, makes sense, I see that at least ActivityPicker uses ```ActivityHandler.toView({number: ....})```, so that place should be updated too if we want to replace ```if(number)``` with ```if(focusComposer)``` inside ```ThreadUI.handleActivity```. So I'm more confident to do this in the master-only follow-up